lib/features/modeling/behavior/CreateChoreoTaskBehavior.js

```diff
@@ -29,144 +29,172 @@ export default function CreateChoreoTaskBehavior(injector, bpmnFactory, canvas,
-      let choreo = canvas.getRootElement();
-      let participants = choreo.businessObject.get('participants');
-
-      // create participants if we do not have two yet
-      while (participants.length < 2) {
-        let participant = bpmnFactory.create('bpmn:Participant');
-        participants.push(participant);
-        participant.name = 'Participant ' + participants.length;
-      }
-      let partA = participants[0];
-      let partB = participants[1];
-
-      // set the properties of the choreo activity business object
-      let activity = shape.businessObject;
-      activity.get('participantRef').push(partA);
-      activity.get('participantRef').push(partB);
-      activity.set('initiatingParticipantRef', partA);
-      activity.set('name', 'New Activity');
-
-      // find out the bounds of the band shapes
-      let boundsA = {
-        x: position.x - shape.width / 2,
-        y: position.y - shape.height / 2,
-        width: shape.width,
-        height: getBandHeight(partA)
-      };
-      let boundsB = {
-        x: position.x - shape.width / 2,
-        y: position.y + shape.height / 2 - getBandHeight(partB),
-        width: shape.width,
-        height: getBandHeight(partB)
-      };
-
-      // create the participant bands
-      let diA = bpmnFactory.createDiShape('bpmndi:BPMNShape', boundsA, {
-        choreographyActivityShape: activity.di,
-        bpmnElement: partA,
-        participantBandKind: 'top_initiating',
-        isMessageVisible: false
-      });
-      let shapeA = elementFactory.createShape(assign({
-        type: 'bpmn:Participant',
-        id: model.ids.nextPrefixed('ParticipantBand_', partA),
-        businessObject: partA,
-        diBand: diA,
-        activityShape: shape
-      }, boundsA));
-
-      let diB = bpmnFactory.createDiShape('bpmndi:BPMNShape', boundsB, {
-        choreographyActivityShape: activity.di,
-        bpmnElement: partB,
-        participantBandKind: 'bottom_non_initiating',
-        isMessageVisible: false
-      });
-      let shapeB = elementFactory.createShape(assign({
-        type: 'bpmn:Participant',
-        id: model.ids.nextPrefixed('ParticipantBand_', partB),
-        businessObject: partB,
-        diBand: diB,
-        activityShape: shape
-      }, boundsB));
-
-      shape.bandShapes = [ shapeA, shapeB ];
-
-      // remember some stuff in the context for later linking
-      assign(context, {
-        createdShapeA: shapeA,
-        createdShapeB: shapeB
-      });
-
-      // create initiating message flow and shape (invisible)
-      if (is(shape, 'bpmn:ChoreographyTask')) {
-        let messageFlow = createMessageFlowSemantics(
-          injector,
-          activity,
-          activity.get('initiatingParticipantRef')
-        );
-        let messageShape = createMessageShape(injector, shapeA, messageFlow);
-        messageShape.hidden = true;
+      //shapes that were created by copy already have participants.
+      if (!shape.oldBusinessObject) {
```

Comment:
Can we add this condition to the outer if to clean up the diff? There don't seem to be any changes except indentation, or am I missing something?